src/Shop.jsx

```diff
@@ -50,16 +53,12 @@ export default function Shop() {
   const effectiveCost = base => Math.max(0, Math.round(base * (1 - discount)));
 
   const buyItem = item => {
-    if (owned[item.id]) return;
+    if (ownedIds.has(item.id)) return;
     const cost = effectiveCost(item.cost);
     if (credits < cost) return;
     setCredits(c => c - cost);
-    setOwned(o => ({ ...o, [item.id]: true }));
-
-    // Simple example effect: voucher applies a discount once.
-    if (item.id === 'co_reroll+') {
-      setDiscount(0.25); // 25% off
-    }
+    addOwnedShopCard(item);
+    if (item.id === 'co_reroll+') setDiscount(0.25);
```

Comment:
Let's not worry about implementing a discount system for now as we have no idea how it will impact the balance of the game.